### PR TITLE
アットナビのhoverのjs勉強用のファイルの追加

### PR DESCRIPTION
**Pull request by seyaa07**
https://www.cosme.net/

グロナビのhoverをやってみる

### CODE REVIEW

File: vol2/css/style.css [hunk 1]
```diff
@@ -0,0 +1,102 @@
+body{
+		padding: 0;
```
Comment (takeuchim) [inline vol2/css/style.css]: paddingとmarginのインデントが1つ深いのと、プロパティの並び順はmarginのほうが先が一般的です。

File: vol2/css/style.css [hunk 2]
```diff
@@ -0,0 +1,102 @@
+body{
+		padding: 0;
+		margin: 0;
+	}
+ul{
```
Comment (takeuchim) [inline vol2/css/style.css]: ulと{の間に半角スペースがあるやつとないやつが混在しています。
どちらかに揃えたほうがいいのと、アイスタイルでは半角スペースあけてるので、開けるほうでそろえましょうか。

File: vol2/css/style.css [hunk 3]
```diff
@@ -0,0 +1,102 @@
+body{
+		padding: 0;
+		margin: 0;
+	}
+ul{
+	padding:0;
```
Comment (takeuchim) [inline vol2/css/style.css]: padding: の後ろに半角スペースがないのでいれましょう。
他でも入っていない部分があるので全体的に見直して修正してください。

File: vol2/css/style.css (hunk 1 above)
Comment (takeuchim) [reply vol2/css/style.css]: プロパティの並び順はちょっと古いけど、これ参考に。

↓の「6.5. プロパティのルール」のところ。
Y:\テクノロジー本部\技術開発部\021_デザインチーム\95_ガイドライン\Guideline\markup\spngen_guide.html

File: vol2/index.html [hunk 4]
```diff
@@ -0,0 +1,52 @@
+<!DOCTYPE html>
+<html><head><meta charset="UTF-8">
+
+<title>jQueryべんきょー2</title>
+
+<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
```
Comment (takeuchim) [inline vol2/index.html]: scriptは表示速度向上のため、body終了タグ手前に移動してください。

File: vol2/index.html [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+</head>
+
+<body>
+<div id="navi">
+	<ul id="g-navi">
+		<li><a href="#">ランキング</a></li>
+		<li><a href="#">ブランド</a></li>
+		<li><a href="#">サロン予約</a></li>
+		<li><a href="#">ブログ</a></li>
+		<li><a href="#">Q&amp;A</a></li>
```
Comment (takeuchim) [inline vol2/index.html:16]: ちゃんと特殊文字使ってるね！

File: vol2/index.html [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+		<li class="js-hover"><a href="#"><span>お買い物</span></a>
+			<div class="js-menu menu">
+				<ul>
+					<li><a href="">アットコスメショッピング（通販）</a></li>
```
Comment (takeuchim) [inline vol2/index.html]: ここのhrefの中身は#がないね。入れるなら入れるで揃えてください。
以下同じところあるので同様に修正してください。

File: vol2/js/jquery.js [hunk 7]
```diff
@@ -0,0 +1,7 @@
+$(function() {
+	$('.js-hover').hover(function(){
```
Comment (takeuchim) [inline vol2/js/jquery.js:2]: このクラスをhoverしたら、メニューを開く閉じるは1つならこれでバッチリ！

複数の場合は、このページが参考になるかも。thisを使って、ホバーされた要素の中のこの要素って指定ができる。
http://semooh.jp/jquery/api/events/hover/over%2C+out/

**Review by takeuchim — CHANGES_REQUESTED** [review]
全部コメントしたら大変な感じになるから、ひとまずぱっと見気になったところを突っ込んでみました。

JSの複数の場合ももう少し頑張ってみて！

File: vol2/index.html [hunk 8]
```diff
@@ -17,15 +15,15 @@
 		<li><a href="#">ブログ</a></li>
 		<li><a href="#">Q&amp;A</a></li>
 		<li class="js-hover"><a href="#"><span>お買い物</span></a>
-			<div class="js-menu menu">
+			<div class="js-menu menu-shopping">
```
Comment (takeuchim) [inline vol2/index.html]: `menu-shopping` と `menu-other` で共通な部分2回CSS書いたりしているから、共通のクラス`memu` を前みたいに設定して、もうひとつそれぞれわけられるように`pt-shopping`かそのまま「memu-shopping」のほうが分かりやすいかな。。。クラス名の設計答えなくてむずし。。。

File: vol2/js/jquery.js (hunk 7 above)
Comment (takeuchim) [reply vol2/js/jquery.js:2]: それぞれ別々に動作するようになったね！！！

File: vol2/index.html (hunk 6 above)
Comment (takeuchim) [reply vol2/index.html]: これなおってないぽい？

File: vol2/css/style.css [hunk 9]
```diff
@@ -1,64 +1,81 @@
...
+.menu-shopping {
+	display: none;
+	background: #796161;
+	position: absolute;
+	top: 35px;
+	right: 0px;
+	min-width: 210px;
+	padding: 12px 0;
+	border: 1px solid #796161;
+	box-shadow: 0px 0px 0px 1px #a18c8b inset,3px 4px 5px 0px rgba(0,0,0,0.3);
```
Comment (takeuchim) [inline vol2/css/style.css]: `0.3` とかの数値は0をとって`.3`にしてください。0なくても動作するはず。１byte節約。あとはGoogleのコーディングガイドラインもこれ

File: vol2/css/style.css [hunk 10]
```diff
@@ -1,64 +1,81 @@
...
+	border: 10px solid transparent;
 	border-top: 10px solid ##5eab99;
 }
-.menu{
+.menu-shopping {
+	display: none;
+	background: #796161;
+	position: absolute;
+	top: 35px;
+	right: 0px;
```
Comment (takeuchim) [inline vol2/css/style.css]: `0`の場合は単位なくてもOKなのでpx削除してください。あと、2byte削減。

File: vol2/css/style.css [hunk 11]
```diff
@@ -1,64 +1,81 @@
...
-#navi{
+a:visited { 
+	color: #068681;
+}
+#navi {
 	background: #eee;
 	border-bottom: 1px solid #d3d3d3;
 	height: 45px;
 	padding-top: 10px;
 	box-shadow:0px -3px 3px -1px #fff inset;
```
Comment (takeuchim) [inline vol2/css/style.css]: `box-shadow:` の後ろに半角スペース入ってないのでいれてください。

File: vol2/css/style.css [hunk 12]
```diff
@@ -1,64 +1,81 @@
...
 }
 a:link { 
 	color: #068681;
 }
-#navi{
+a:visited { 
+	color: #068681;
+}
+#navi {
 	background: #eee;
```
Comment (takeuchim) [inline vol2/css/style.css]: プロパティの並び順が、borderとbackgroundはもっと下のはず。並び順を確認してみて！

File: vol2/css/style.css [hunk 13]
```diff
@@ -1,64 +1,81 @@
...
 	border-bottom: 1px solid #d3d3d3;
 	height: 45px;
 	padding-top: 10px;
 	box-shadow:0px -3px 3px -1px #fff inset;
 }
-#g-navi{
+#g-navi {
+	display: block;
```
Comment (takeuchim) [inline vol2/css/style.css]: こことかもプロパティの並び順が違うので、がんばって並び順見てみて！
以下のセレクタに関しても同様でっす！
並び順大変なんだけど、複数人で作業するときはメンテナンス性とか作業効率にもかかわってくるからがんば！

**Review by takeuchim — COMMENTED** [review]
@seyaa07 ばっちりJS動いたね！あと、細かい部分なんだけど、コーディングルール的なところで気になったところあったからコメントしますた！

File: vol2/css/style.css [hunk 14]
```diff
@@ -0,0 +1,114 @@
...
+}
+a:link { 
+	color: #068681;
+}
+a:visited { 
+	color: #068681;
+}
+#navi {
+	padding-top: 10px;
+	height: 45px;
```
Comment (takeuchim) [inline vol2/css/style.css:19]: おしい、heightのほうがpaddingより上やで！

File: vol2/css/style.css [hunk 15]
```diff
@@ -0,0 +1,114 @@
...
+}
+#g-navi > li:last-child > a {
+	border-right: none;
+}
+
+#g-navi span {
+	position: relative;
+}
+#g-navi span::after {
+	content: "";
```
Comment (takeuchim) [inline vol2/css/style.css:56]: contentのほうが下のほうだった気がする！
あと、ダブルクオーテーションじゃなくて、シングル使ってね！
JSに関してもシングルで文字列とか囲うからそれと揃えて。JSでシングル使う理由としては、文字列でHTML扱うことがあって、ダブルだと属性のい囲みでエスケープしないといけない。シングルならそれ不要なので。

File: vol2/css/style.css [hunk 16]
```diff
@@ -0,0 +1,114 @@
...
+	display: block;
+	position: absolute;
+	top: 10px;
+	right: 10px;
+	width: 0;
+	height: 0;
+	border: 10px solid transparent;
+	border-top: 10px solid ##5eab99;
```
Comment (takeuchim) [inline vol2/css/style.css:64]: `##5eab99` が# 2個あるけど、色大丈夫？？

File: vol2/css/style.css [hunk 17]
```diff
@@ -0,0 +1,114 @@
...
+}
+.menu{
+	display: none;
+	position: absolute;
+	top: 35px;
+	right: 0;
+	padding: 12px 0;
+	border: 1px solid #796161;
+	background: #796161;
+	box-shadow: 0px 0px 0px 1px #a18c8b inset,3px 4px 5px 0px rgba(0,0,0,.3);
```
Comment (takeuchim) [inline vol2/css/style.css:74]: ここも0の場合はpxなしで！

File: vol2/css/style.css [hunk 18]
```diff
@@ -0,0 +1,114 @@
...
+}
+.pt-shopping {
+	min-width: 210px;
+}
+.pt-other {
+	min-width: 325px;
+}
+.pt-shopping::before,
+.pt-other::before {
+	content:'';
```
Comment (takeuchim) [inline vol2/css/style.css:84]: contentの位置を下に！

File: vol2/css/style.css (hunk 15 above)
Comment (takeuchim) [reply vol2/css/style.css:56]: Y:\テクノロジー本部\デザイン部\95_ガイドライン\Guideline\markup\spn\gen_guide.html
の6.5. プロパティのルールの6.「フォント」「テキスト」「段組み」に関するプロパティの中に入ってるよ～。

**Review by takeuchim — COMMENTED** [review]
@seyaa07 あともちょっとコーディングルール系の微調整お願いしま！
